datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -108,6 +110,8 @@ impl AggregateMode {
     }
 }
 
+const INTERNAL_GROUPING_ID: &str = "grouping_id";
```

Comment:
Seems to just work (which surprised me).

Reply:
Pushed changes to add it to the logical plan, so now name conflicts are possible. And renamed it `__grouping_id`, this follows the pattern used for `__common_expr` which can also conflict with names from user schema. Since it it internal I think it should be possible to follow up and make it fully unique.